examples/amm/lib.rs

```diff
@@ -0,0 +1,570 @@
+//! ## Automated Money Market (AMM)
+//!
+//! An AMM is a decentralized marketplace to swap tokens. Usually swap fees are significantly cheaper than centralized exchanges,
+//! and this system allows anyone to participate as either the Swapper (a user interested in swapping one token for another)
+//! or as a Liquidity Provide (LP), //! who is financially incentivised to deposit equal amounts of token1 and token2. Each time
+//! a token is swapped a small percentage is given to all the LPs, and each share is proportional to the amount of liquidity
+//! that a LP provides. Pretty simple, right? You can learn more about AMMs [here](https://www.coindesk.com/learn/2021/08/20/what-is-an-automated-market-maker/).
+//!
+//! This code example was inspired by [this blog](https://learn.figment.io/tutorials/build-polkadot-amm-using-ink).
+//!
+//! ## Warning
+//!
+//! This contract is an *example*. It is neither audited nor endorsed for production use.
+//! Do **not** rely on it to keep any
+//!
+//! ## Tokens and AMMs
+//!
+//! If you are familiar with Solidity you may have worked with AMM contracts that integrate multiple ERC20 contracts for
+//! token pairs. This is a composable contract model. In this code example, however, we demonstrate a more simple version where all tokens
+//! are maintained by this contract (one contract instead of many composed contracts).
+//!
+//! ## The Faucet
+//!
+//! There is a `faucet()` function built in to this contract to fund your wallet balance. You can use this to //! test dive the features
+//! in a staging environment, and in tests.
```

Comment:
Can you also add a section about the workflow and usage? E.g, first you must add liquidity to a pool, then you can swap, et.c